libbpf-rs/src/link.rs

```diff
@@ -61,6 +61,15 @@ impl Link {
             Ok(())
         }
     }
+
+    /// Get file descriptor of link.
+    ///
+    /// # Safety
+    ///
+    /// It is not safe to use the file descriptor after this link is dropped.
+    pub unsafe fn get_fd(&self) -> i32 {
```

Comment:
Should be ok to leave this safe just like map API: https://docs.rs/libbpf-rs/0.7.2/libbpf_rs/struct.Map.html#method.fd. Safe is more in line with rest of rust. For example, you can take pointer of anything safely but when you try to deref it it'll involve unsafety. See my note above about how we'll generally make an API unsafe if we take in a FD.